students/donghwa/users/views.py

```diff
@@ -1,3 +1,44 @@
 from django.shortcuts import render
 
 # Create your views here.
+
+import json, re
+
+from django.http     import JsonResponse
+from django.views    import View
+
+from users.models    import User
+
+class SignUpView(View):
+    def post(self, request):
+
+        try:
+            data = json.loads(request.body)
+
+
+            regexp_email    = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+            regexp_password = re.compile('^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$')
+
+            if re.match(regexp_email, data['email']) is None:
+                return JsonResponse({'message' : 'WRONG_EMAIL_FORMAT'}, status=400)
+
+            if re.match(regexp_password, data['password']) is None:
+                return JsonResponse({'message' : 'WRONG_PASSWORD_FORMAT'}, status=400)
```

Comment:
파이썬에서는 `if 조건문` 평가시, 반환 되는 값 자체가 `boolean` 일 경우나, `Truthy` `Falsy` 한 값일 경우,  `== False` `== None` `== 0`등으로 평가하지 않고, 그 값을 그대로 사용합니다. 그 자체로도 `True or False`를 판단할 수 있기 때문입니다. `False`한 지 평가하실 때는 `not` 을 활용해보세요 :)

```suggestion
            if not re.match(regexp_email, data['email']):
                return JsonResponse({'message' : 'WRONG_EMAIL_FORMAT'}, status=400)

            if not re.match(regexp_password, data['password']):
                return JsonResponse({'message' : 'WRONG_PASSWORD_FORMAT'}, status=400)
```